dashboard/app/assets/javascripts/sendReport.js

```diff
@@ -41,6 +47,9 @@ var sendReport = function(report) {
     type: 'POST',
     url: report.callback,
     contentType: 'application/x-www-form-urlencoded',
+    // Set a timeout of a few seconds so the user will get the fallback
+    // response even if the server is hung and unresponsive.
+    timeout: 4000,
```

Comment:
In practice, a 4 second timeout seems very aggressive. I'd expect something closer to 15 seconds to deal with real-world network latencies on slow networks (3G, LTE for mobile devices)